src/crawler/getPolkadotData.ts

```diff
@@ -22,130 +23,73 @@ export async function runCrawlers() {
 
-        // As Multi
-        const asMultiBlock = await api.derive.chain.getBlock(blockHash);
-        let asMulti = asMultiBlock?.extrinsics.filter((ex) => {
-            return ex.extrinsic.method.section == 'multisig' && ex.extrinsic.method.method == 'asMulti';
-        });
-        asMulti && asMulti.map(async (singleEx) => {
-            singleEx.events.map((singleEvent) => {
-                if (singleEvent.section == 'multisig') {
-                    // let hash = blake2AsHex(singleEx.extrinsic.method.args[3].toHuman()?.toString()!);
-                    let multisigDBRecord: IAsMulti_MultiSigWallet = {
-                        address: singleEvent.method == 'MultisigExecuted' ? singleEvent.data[2].toHuman()?.toString()! : singleEvent.data[1].toHuman()?.toString()!,
-                        signature: singleEx.extrinsic.signature.toHuman()?.toString()!,
-                        signer: singleEx.extrinsic.signer.toHuman()?.toString()!,
-                        signitories: (singleEx.extrinsic.method.args[1] as any).map((signitory) => {
-                            return String(signitory.toHuman())
-                        }),
-                        method: singleEx.extrinsic.method.method,
-                        eventType: singleEvent.method == 'MultisigExecuted' ? 'MultisigExecuted' : 'NewMultisig',
-                        callHash: blake2AsHex(singleEx.extrinsic.method.args[3].toHuman()?.toString()!),
-                        callData: singleEx.extrinsic.method.args[3].toHuman()?.toString()!,
-                        threshold: singleEx.extrinsic.method.args[0].toHuman()?.toString()!,
-                        tip: singleEx.extrinsic.tip.toHuman()?.toString()!,
-                        maybeTimepoint: singleEx.extrinsic.method.args[2].toHuman(),
-                        maxWeight: singleEx.extrinsic.method.args[5].toHuman()
+                    payload.multisig_address = event.method == 'NewMultisig' ? event.data[1].toHuman()?.toString()! : event.data[2].toHuman()?.toString()!;
+
+                    payload.call_hash = _datumPair.extrinsic.method.method == 'asMulti' ? blake2AsHex(_datumPair.extrinsic.method.args[3].toHuman()?.toString()!) : _datumPair.extrinsic.method.args[3].toHuman()?.toString()!; // only as_multi will have actual call_data other calls will only have the call_hash
+                    // for final approval we send full call data instead of the hash
+
+                    payload.call_data = _datumPair.extrinsic.method.method == 'asMulti' ? _datumPair.extrinsic.method.args[3].toHuman()?.toString()! : (await storage.query({ "item.call_hash": _datumPair.extrinsic.method.args[3].toHuman()?.toString()! }) as Array<any>)[0]?.item.call_data; // We don't get call_data until the final approval call (as_multi), so I think the call_data for other calls would be empty. but in case we had a call_data in DB before from some other final call of as_multi then we'd have its data and we'd add it here.
```

Comment:
If you want to query the call_data from storage, you need to consider the timepoint of the storage.

Reply:
Would it matter what timepoint the call_data is taken from if the hash is same as the current one? I think hash collision is highly unlikely in this scenario?